File: src/polarssl.cc
#include <node.h>

#include "hash.h"
#include "hmac.h"
#include "keygen.h"
#include "key-rsa.h"
#include "random.h"

void InitAll(v8::Handle<v8::Object> exports) {
  PolarSSL::Hash::Init(exports);
  PolarSSL::HMAC::Init(exports);
  PolarSSL::Keygen::Init(exports);
  PolarSSL::KeyRSA::Init(exports);
  PolarSSL::Random::Init(exports);
}

NODE_MODULE(polarssl, InitAll)

File: polarssl.js
var _polarssl = require("./build/Release/polarssl");

var polarssl = module.exports = {
  Cipher: require("./lib/cipher"),
  Decipher: require("./lib/decipher"),
  getCiphers: _polarssl.getCiphers,
  Hash: require("./lib/hash"),
  HMAC: require("./lib/hmac"),
  getHashes: _polarssl.getHashes,
  Keygen: _polarssl.Keygen,
  KeyRSA: _polarssl.KeyRSA,
  Random: _polarssl.Random,
};

polarssl.createCipher = function createCipher(name, key, iv) {
  return new polarssl.Cipher(name, key, iv);
};

polarssl.createDecipher = function createDecipher(name, key, iv) {
  return new polarssl.Decipher(name, key, iv);
};

polarssl.createHash = function createHash(name) {
  return new polarssl.Hash(name);
};

polarssl.createHmac = function createHmac(name, key) {
  return new polarssl.HMAC(name, key);
};

polarssl.createKeygen = function createKeygen() {
  return new polarssl.Keygen();
};
